### DISCUSSION TITLE
How to save a user's filtered query to supabase?

### DISCUSSION THREAD

**Discussion by zlever01** [supabase/supabase#3080] [Questions] [answered]
New/hobby developer here. I'm working on a next.js app where users can search and filter through products. My supabase query looks something like this: 

```
const {data, count} = await supabase
        .from('product')
        .select('*', {count: "exact"})
        .eq(brandType, brandValue)
        .or(colorArray)
        .or(typeArray)
        .gte('weight', minString)
        .lte('weight', maxString)
        .order(sortValue[0], {ascending: sortValue[1]})
        .range(from, to)
```

How can I have users save their searches? How can I insert a query into a user's table for future reference/display data in the user's account? 

With that, is there a way to email users when new products match their saved queries? I can't tell if on().subscribe() can handle that if the user is not actively using the site.

**Comment by soedirgo** [comment, marked as answer]
> How can I have users save their searches? How can I insert a query into a user's table for future reference/display data in the user's account?

Hey @zlever01! I remember this being asked before, but unfortunately there isn't anything from the client library that makes this easy.

That said, you may be able to do this by dynamically adding the filters. As a minimal example:
```js
const { createClient } = require('@supabase/supabase-js')
let supabase = createClient(
  '...',
  '...'
)
const filters = [['gt', 'id', 1], ['lt', 'id', 3]]
supabase = filters
  .reduce(
    (acc, [filter, ...args]) => {
      return acc[filter](...args)
    },
    supabase
      .from('my_table')
      .select(),
  )
supabase.then(console.log)
```
```js
{
  ...,
  data: [
    {
      id: 2
    }
  ],
  ...
}
```

Then you can store `filters` in a `jsonb` column in, say, your `users` table.

> With that, is there a way to email users when new products match their saved queries?

I was about to say webhooks (e.g. with the experimental Supabase Hooks) would be a better fit, but since your filters are defined in terms of the client library, you'll need to run some JS to apply it, so unfortunately this isn't easy to do either.

**Reply by zlever01** [reply]
Thanks @soedirgo!
